packages/core/android/src/main/kotlin/com/datadog/reactnative/DdSdk.kt

```diff
@@ -281,6 +287,34 @@ class DdSdk(
         return configBuilder.build()
     }
 
+    private fun buildFirstPartyHosts(
+        firstPartyHosts: List<ReadableMap>
+    ): Map<String, Set<TracingHeaderType>> {
+        /**
+         * Adapts the data format from the React Native SDK configuration to match with the
+         * Android SDK configuration. For example:
+         *
+         * RN config: [{ match: "example.com", propagatorType: [DATADOG, B3] }]
+         * Android config: { "example.com": [DATADOG, B3] }
+         */
+        val firstPartyHostsWithHeaderType = mutableMapOf<String, MutableSet<TracingHeaderType>>()
```

Comment:
minor: can be just `Set` here and then below it is enough to have just `propagatorTypes.toSet()`

```suggestion
        val firstPartyHostsWithHeaderType = mutableMapOf<String, Set<TracingHeaderType>>()
```

Reply:
I believe this does not work because we want to sometimes do `addAll` to the sets. It can be possible that the same `"match"` (host) is declared multiple times in javascript.
I will add a test for this case though

Reply:
indeed, I missed that